src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Initialize/InitializeHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.LanguageServer.Handler
+{
+    [Shared]
+    [ExportLspMethod(Methods.InitializeName, StringConstants.XamlLanguageName)]
+    internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeResult>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InitializeHandler()
+        {
+        }
+
+        public Task<InitializeResult> HandleRequestAsync(InitializeParams request, ClientCapabilities clientCapabilities, string? clientName, CancellationToken cancellationToken)
+        {
+
+            return Task.FromResult(new InitializeResult
+            {
+                Capabilities = new ServerCapabilities
+                {
+                    CompletionProvider = new CompletionOptions { ResolveProvider = true, TriggerCharacters = new string[] { "<", " ", ":", ".", "=", "\"", "'", "{", ",", "(" } },
+                    HoverProvider = true,
+                    FoldingRangeProvider = new FoldingRangeProviderOptions { },
+                    TextDocumentSync = new TextDocumentSyncOptions
+                    {
+                        Change = TextDocumentSyncKind.None
```

Comment:
Note that we will be changing this soon to use LSP document sync to enable classifications support.  Would it be problematic for you if XAML also used LSP for document synchronization?

The basic idea is to save document text in the server from LSP didChange (for all open documents) and create a forked solution with base of workspace. CurrentSolution + text from LSP open documents